pkg/api/steve/proxy/path_test.go

```diff
@@ -0,0 +1,110 @@
+package proxy
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func TestGVRFromPath(t *testing.T) {
```

Comment:
Personally, I would not recommend testing this function directly. It's a non-exported implementation detail used by a single function (the `matchManagementCRDs` function). I think all of the test cases here should really be part of testing the matching/auth function.

Reply:
Agreed, I am using these tests as a safety net during dev, so can remove when we're closer to merging.
On the other hand, this function with its tests might be useful for us in the future (if exported and placed in some other package), since we haven't found an equivalent in any K8s SDKs.

Reply:
If we export it I think it's definitely worth testing.

That being said, we can't rely on the namespaces part if we export this. I think it works for us since management api group doesn't have a namespaces type - but other api groups might have a namespace type. Since this flaw exists (and there's no exported check for the error message, etc.), I think this definitely falls into the category of implementation detail.

Reply:
I see writing tests for exported functions only as a guideline, not a rule.

I like having these tests. It isn't required, but it helps to have tests for complicated functions like this, which can break easily if alternated in the future.